src/hooks/useIsInRole.ts

```diff
@@ -0,0 +1,19 @@
+import { useAuthContext } from "../context/AuthContext";
+
+/**
+ * True if the user is logged in and in that role.
+ * @param role The required role.
+ */
+export const useIsInRole = (role: string) => {
```

Comment:
Maybe this could be a component that checks for the role. Additionally it could show either an unauthorized screen or nothing at all

Reply:
can't use components in hooks, but hooks in components